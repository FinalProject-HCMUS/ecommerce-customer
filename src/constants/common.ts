const orderStatus = [
  { value: 'all', label: 'All Orders' },
  { value: 'NEW', label: 'New' },
  { value: 'PROCESSING', label: 'Processing' },
  { value: 'PACKAGED', label: 'Packaged' },
  { value: 'PICKED', label: 'Picked' },
  { value: 'SHIPPING', label: 'Shipping' },
  { value: 'DELIVERED', label: 'Delivered' },
  { value: 'CANCELLED', label: 'Cancelled' },
  { value: 'REFUNDED', label: 'Refunded' },
];

const TOP_PRODUCTS_PER_PAGE = 100;
const PRODUCT_PER_PAGE = 18;
const BLOG_PER_PAGE = 10;
const TOP_TRENDING = 'TOP TRENDING';
const TOP_SELLING = 'TOP SELLING';
const VISIBLE_PRODUCT = 18;

const COLOR_PER_PAGE = 100;
const SIZE_PER_PAGE = 10;
const CATEGORY_PER_PAGE = 10;

const TIME_OUT_ADD_TO_CART = 2000;
const DESCREASE = 'decrease';
const INCREASE = 'increase';

const MESSAGE_PER_PAGE = 100;

const PAYMENT_METHOD = [
  { id: 'COD', label: 'COD' },
  { id: 'VN_PAY', label: 'Vn Pay' },
];

export default {
  orderStatus,
  TOP_PRODUCTS_PER_PAGE,
  PRODUCT_PER_PAGE,
  BLOG_PER_PAGE,
  TOP_SELLING,
  TOP_TRENDING,
  VISIBLE_PRODUCT,
  COLOR_PER_PAGE,
  SIZE_PER_PAGE,
  CATEGORY_PER_PAGE,
  TIME_OUT_ADD_TO_CART,
  DESCREASE,
  INCREASE,
  PAYMENT_METHOD,
  MESSAGE_PER_PAGE,
};
