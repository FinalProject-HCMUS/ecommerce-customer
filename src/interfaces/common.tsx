export interface IRoute {
    exact?: boolean;
    path: string;
    name: string;
    component?: React.ElementType;
    children?: string[];
    flagKey?: string;
    root?: string;
}