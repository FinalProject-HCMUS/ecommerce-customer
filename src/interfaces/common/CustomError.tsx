export interface CustomError {
  timestamp: string;
  httpStatus: string;
  header: string;
  message?: string;
  isSuccess: boolean;
  exceptionName?: string;
  subsErrors?: CustomSubError[];
}

export interface CustomSubError {
  message: string;
  field: string;
  value?: object;
  type?: string;
}
