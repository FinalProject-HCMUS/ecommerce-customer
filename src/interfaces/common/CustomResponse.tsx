export interface CustomResponse<T> {
  /**
   * Timestamp of the response
   * @example "2023-01-01T12:00:00"
   */
  timestamp: string

  /**
   * HTTP status of the response
   * @example "OK"
   */
  httpStatus: string

  /**
   * Indicates if the operation was successful
   * @example true
   */
  isSuccess: boolean

  /**
   * Response data (null for operations without return data)
   */
  data?: T
}
