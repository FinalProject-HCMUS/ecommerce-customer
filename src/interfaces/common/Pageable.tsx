export interface Sort {
  sorted: boolean
  unsorted: boolean
  empty: boolean
}
export interface PageableDetails {
  paged: boolean
  pageNumber: number
  pageSize: number
  unpaged: boolean
  offset: number
  sort: Sort
}

export interface Pageable<T> {
  totalElements: number
  totalPages: number
  last: boolean
  first: boolean
  pageable: PageableDetails
  numberOfElements: number
  size: number
  content: T
  number: number
  sort: Sort
  empty: boolean
}
