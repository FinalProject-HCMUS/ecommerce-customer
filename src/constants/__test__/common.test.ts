import common from '../common';

describe('common constants', () => {
  test('should export an object with all expected properties', () => {
    expect(common).toBeDefined();
    expect(typeof common).toBe('object');

    // Check that all expected properties exist
    const expectedProps = [
      'orderStatus',
      'TOP_PRODUCTS_PER_PAGE',
      'PRODUCT_PER_PAGE',
      'BLOG_PER_PAGE',
      'TOP_TRENDING',
      'TOP_SELLING',
      'VISIBLE_PRODUCT',
      'COLOR_PER_PAGE',
      'SIZE_PER_PAGE',
      'CATEGORY_PER_PAGE',
      'TIME_OUT_ADD_TO_CART',
      'DESCREASE',
      'INCREASE',
      'PAYMENT_METHOD',
      'MESSAGE_PER_PAGE',
    ];

    expectedProps.forEach((prop) => {
      expect(common).toHaveProperty(prop);
    });
  });

  test('orderStatus should contain all expected statuses', () => {
    expect(Array.isArray(common.orderStatus)).toBe(true);
    expect(common.orderStatus).toHaveLength(9);

    // Check specific entries
    expect(common.orderStatus).toContainEqual({
      value: 'all',
      label: 'All Orders',
    });
    expect(common.orderStatus).toContainEqual({ value: 'NEW', label: 'New' });
    expect(common.orderStatus).toContainEqual({
      value: 'PROCESSING',
      label: 'Processing',
    });
    expect(common.orderStatus).toContainEqual({
      value: 'DELIVERED',
      label: 'Delivered',
    });
    expect(common.orderStatus).toContainEqual({
      value: 'CANCELLED',
      label: 'Cancelled',
    });

    // Check structure of each entry
    common.orderStatus.forEach((status) => {
      expect(status).toHaveProperty('value');
      expect(status).toHaveProperty('label');
      expect(typeof status.value).toBe('string');
      expect(typeof status.label).toBe('string');
    });
  });

  test('pagination constants should have correct values', () => {
    expect(common.TOP_PRODUCTS_PER_PAGE).toBe(100);
    expect(common.PRODUCT_PER_PAGE).toBe(18);
    expect(common.BLOG_PER_PAGE).toBe(10);
    expect(common.VISIBLE_PRODUCT).toBe(18);
    expect(common.COLOR_PER_PAGE).toBe(100);
    expect(common.SIZE_PER_PAGE).toBe(10);
    expect(common.CATEGORY_PER_PAGE).toBe(10);
    expect(common.MESSAGE_PER_PAGE).toBe(100);
  });

  test('string constants should have correct values', () => {
    expect(common.TOP_TRENDING).toBe('TOP TRENDING');
    expect(common.TOP_SELLING).toBe('TOP SELLING');
    expect(common.DESCREASE).toBe('decrease');
    expect(common.INCREASE).toBe('increase');
  });

  test('TIME_OUT_ADD_TO_CART should be a number', () => {
    expect(typeof common.TIME_OUT_ADD_TO_CART).toBe('number');
    expect(common.TIME_OUT_ADD_TO_CART).toBe(2000);
  });

  test('PAYMENT_METHOD should contain all expected payment methods', () => {
    expect(Array.isArray(common.PAYMENT_METHOD)).toBe(true);
    expect(common.PAYMENT_METHOD).toHaveLength(2);

    // Check specific entries
    expect(common.PAYMENT_METHOD).toContainEqual({ id: 'COD', label: 'COD' });
    expect(common.PAYMENT_METHOD).toContainEqual({
      id: 'VN_PAY',
      label: 'Vn Pay',
    });

    // Check structure of each entry
    common.PAYMENT_METHOD.forEach((method) => {
      expect(method).toHaveProperty('id');
      expect(method).toHaveProperty('label');
      expect(typeof method.id).toBe('string');
      expect(typeof method.label).toBe('string');
    });
  });

  test('should match snapshot', () => {
    expect(common).toMatchSnapshot();
  });
});
