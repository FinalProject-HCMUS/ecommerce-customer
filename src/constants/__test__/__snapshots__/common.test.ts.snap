// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common constants should match snapshot 1`] = `
{
  "BLOG_PER_PAGE": 10,
  "CATEGORY_PER_PAGE": 10,
  "COLOR_PER_PAGE": 100,
  "DESCREASE": "decrease",
  "INCREASE": "increase",
  "MESSAGE_PER_PAGE": 100,
  "PAYMENT_METHOD": [
    {
      "id": "COD",
      "label": "COD",
    },
    {
      "id": "VN_PAY",
      "label": "Vn Pay",
    },
  ],
  "PRODUCT_PER_PAGE": 18,
  "SIZE_PER_PAGE": 10,
  "TIME_OUT_ADD_TO_CART": 2000,
  "TOP_PRODUCTS_PER_PAGE": 100,
  "TOP_SELLING": "TOP SELLING",
  "TOP_TRENDING": "TOP TRENDING",
  "VISIBLE_PRODUCT": 18,
  "orderStatus": [
    {
      "label": "All Orders",
      "value": "all",
    },
    {
      "label": "New",
      "value": "NEW",
    },
    {
      "label": "Processing",
      "value": "PROCESSING",
    },
    {
      "label": "Packaged",
      "value": "PACKAGED",
    },
    {
      "label": "Picked",
      "value": "PICKED",
    },
    {
      "label": "Shipping",
      "value": "SHIPPING",
    },
    {
      "label": "Delivered",
      "value": "DELIVERED",
    },
    {
      "label": "Cancelled",
      "value": "CANCELLED",
    },
    {
      "label": "Refunded",
      "value": "REFUNDED",
    },
  ],
}
`;
